desc/integrals/interp_utils.py

```diff
@@ -9,10 +9,438 @@
 
 from functools import partial
 
+import numpy as np
 from interpax import interp1d
+from orthax.chebyshev import chebroots
+
+from desc.backend import dct, jnp, rfft, rfft2, take
+from desc.integrals.quad_utils import bijection_from_disc
+from desc.utils import Index, errorif, safediv
+
+# TODO: Boyd's method 𝒪(N²) instead of Chebyshev companion matrix 𝒪(N³).
+#  John P. Boyd, Computing real roots of a polynomial in Chebyshev series
+#  form through subdivision. https://doi.org/10.1016/j.apnum.2005.09.007.
+#  Use that once to find extrema of |B| if N_B > 64. Then to find roots
+#  of bounce points use the closed formula in Boyd's spectral methods
+#  section 19.6. Can isolate interval to search for root by observing
+#  whether B - 1/pitch changes sign at extrema. This is significantly
+#  cheaper and non-iterative, so jax and gpu will like it.
+chebroots_vec = jnp.vectorize(chebroots, signature="(m)->(n)")
+
+
+# TODO: Transformation to make nodes more uniform Boyd eq. 16.46 pg. 336.
+#  More uniformly spaced nodes might speed up convergence.
```

Comment:
at the extreme could do uniformly spaced and just extend domain to [0 - delta, 2pi + delta] and ignore the delta region where runge is isolated, and use the next cut of the field line